pkg/releaselock/rpc.go

```diff
@@ -86,23 +88,31 @@ func ReleaseLock(hostIP, clientIP string) error {
 	}()
 
 	// Create client using sunrpc codec.
-	client := rpc.NewClientWithCodec(sunrpc.NewClientCodec(conn, notifyClose))
+	client := sunrpc.NewClientCodec(conn, notifyClose)
 
 	klog.Infof("Calling Filestore IP %s to release all locks for GKE node %s", serverAddress, clientIP)
 
-	ipByte := net.ParseIP(clientIP).To4()
-	if ipByte == nil {
+	ip := net.ParseIP(clientIP)
+	if ip == nil {
 		return fmt.Errorf("invalid GKE node IP %s", clientIP)
 	}
+	ipByte := ip.To4()
+	if ipByte == nil {
+		return fmt.Errorf("invalid GKE node IPv4 address %s", clientIP)
+	}
 	ipBinary := binary.BigEndian.Uint32(ipByte)
 
-	var response releaseLockResponse
-	if err := client.Call(inbandLockReleaseProcedureName, ipBinary, response /* empty response body */); err != nil {
-		return fmt.Errorf("failed to call procedure %s for GKE node IP %s Filestore IP %s, err: %w", inbandLockReleaseProcedureName, clientIP, hostIP, err)
+	request := rpc.Request{
```

Comment:
I tried to decouple `WriteRequest()` and `ReadResponseHeader()` from `client.Call()`. 
The reasons are: 
1. We have direct control of `seq`, so that each rpc call from client will have a unique `xid`.
2. The channel in client.Call() will never close for some reason if we modify `req` in `WriteRequest()`. This will cause the rpc call getting blocked.

The cons of this approach is that this rpc call is synchronized. But it should be fine since we have only one service method (`IN_BAND_PROPRIETARY_LOCK_OPS_V1`). Besides, we'll make the lock release logic as a controller model, if the RPC call failed, it will be called again in the next reconcile loop.